### ISSUE TITLE
Blank white screen at launch on Windows 10

### ISSUE THREAD

**Issue by aochagavia** [servo/servo#13257]
# Cause

**EDIT**: This is caused by https://github.com/servo/servo/issues/13371 (font rendering is not implemented for the MSVC toolchain on Windows)

# Reproducing

Run `mach.bat run -d https://www.mozilla.org/en-US/`

# Screenshot

![Screenshot](https://cloud.githubusercontent.com/assets/5196584/18475493/e621b3e4-79c6-11e6-93ee-06ddeee2e9c5.png)

**Comment by Coder206** [comment]
@aochagavia Another fact I noticed is that you can reproduce simply with `mach.bat run -d`

**Comment by gaby2300** [comment]
I have a similar issue. I get a white blank window that includes blue related to servo links for a few seconds and then goes white again. The title is Servo, the parallel browser engine

**Comment by CAfromCA** [comment]
Some of these may be issues rendering the servo.org home page, which I am also seeing. The page renders for a split second, then appears to go white.

If I scroll far enough, though, the "Contributing" section appears as does the rest of the page below that. I can also move my mouse around in the blank area until the pointer changes to find and click on invisible links like the download button.

@gaby2300: What happens if you use your mouse wheel or arrow keys to scroll down?

**Comment by gaby2300** [comment]
@CAfromCA I get the same results as you. Clicking the Start Contributing button leads me to a Servo Starters site and a list of open issues. Clicking an invisible link at the top where the pointer changes to a hand lead me to a screenshot titled "browser.html - Servo". "Servo, the parallel browser" and a blue download button to for a Servo nightly build inside are part of the screenshot's content.

In all cases there was no way to go back: no tabs, no context menu.

**Comment by larsbergstrom** [comment]
The problem is that, with the MSVC  toolchain, we are by default currently not able to find the fonts:
```
C:\Users\lars\servo>target\debug\servo.exe
Unable to find any fonts that match (do you have fallback fonts installed?) (thread FontCacheThread, at C:\Users\lars\servo\components\gfx\font_cache_thread.rs:393)
ERROR:servo: Unable to find any fonts that match (do you have fallback fonts installed?)
failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) }) (thread LayoutWorker worker 1/3, at ../src/libcore\result.rs:799)
failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) }) (thread LayoutWorker worker 2/3, at ../src/libcore\result.rs:799)
ERROR:servo: failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) })
ERROR:servo: failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) })
failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) }) (thread LayoutWorker worker 3/3, at ../src/libcore\result.rs:799)
ERROR:servo: failed to receive response to font request: IoError(Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) })
called `Result::unwrap()` on an `Err` value: "SendError(..)" (thread LayoutThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }, at ../src/libcore\result.rs:799)
ERROR:servo: called `Result::unwrap()` on an `Err` value: "SendError(..)"
called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }" (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }, at ../src/libcore\result.rs:799)
ERROR:servo: called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"
failed to send message to font cache thread: Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) } (thread LayoutWorker worker 1/3, at ../src/libcore\result.rs:799)
ERROR:servo: failed to send message to font cache thread: Error { repr: Custom(Custom { kind: BrokenPipe, error: StringError("MPSC channel closed") }) }
```

In order to get errors in the console, of course, after doing a build you need to do `editbin /SUBSYSTEM:CONSOLE target\debug\servo.exe`

cc @metajack

**Comment by aochagavia** [comment]
The problem is in `components/gfx/platform/mod.rs`

```
#[cfg(all(target_os = "windows", target_env = "msvc"))]
pub use platform::dummy::{font, font_context, font_list, font_template};
```

**Comment by CAfromCA** [comment]
Should this be closed as a duplicate of #13371, since the lack of font rendering on Windows seems to be the underlying cause for the reports of blank Servo windows?

**Comment by mbrubeck** [comment]
Closing as duplicate.
